src/main.ts

```diff
@@ -85,6 +87,11 @@ async function installRustUp(): Promise<void> {
 }
 
 async function runCargoSemverChecks(cargo: rustCore.Cargo): Promise<void> {
+    // The default location of the target directory varies depending on whether
+    // the action is run inside a workspace or on a single crate. We therefore
+    // need to set the target directory explicitly.
+    process.env["CARGO_TARGET_DIR"] = "target";
```

Comment:
Did you test to make sure this does the right thing?

Internally, `cargo-semver-checks` uses [an explicit `--target-dir` option](https://github.com/obi1kenobi/cargo-semver-checks/blob/fe8ea52b8c461f433ff49f1c7ae47bc481689455/src/rustdoc_cmd.rs#L68-L79) when generating rustdoc, and I believe the explicit CLI option overrides the env var.

This would be great to have tests for, since it's something we could very easily accidentally break with serious consequences for our users.

Reply:
This is covered by `Test action inputs / Test input manifest-path against path containing space`: the `cargo-semver-checks` is run outside any workspace, with `--manifest-path="ref slice/Cargo.toml"`. Then, the default location would be `./ref slice/target/`, but it is being saved in `./target/`, as it should be. Otherwise, the test would fail, because `saveCache` throws an error if the supplied path does not exist.  

Do you think we can do any better? I'd be great to assert if the cache is being restored successfully (for example by looking for `baseline, cached` in the stdout), but as far as I know it is currently impossible to capture stdout of the whole action as a step in a workflow, so it would require changes in the action itself...

Reply:
Oh, well, I just realized that relying on `saveCache` error is quite stupid at the moment, because it was called only once in the beginning as a result of the `if` statement we are discussing below... Removing the `if` will fix this issue, but maybe you have some better idea of how to test the caching?

Reply:
In the test workflow, could we just check after running the workflow that the target cache directory contains the files we expect and that they are not empty?

I'm less worried about the caching logic not uploading the files, and more worried about for example:
- a repeat of the bug where the baseline JSON is overwritten with the current JSON, where the cache directory ends up empty
- a `--workspace` run saving only the first / only the last package's baseline in the cache, and not the others

Also perhaps consider making the directory a bit more unique to our action. Our action shares the filesystem with everything else running in the same job, so it's possible that some other action also does something at `./target/` and may have an unfortunate interaction we may need to debug. Perhaps use `./semver-checks/target/` instead?

Zooming way out, here's why we're doing all this: we expect tons of people and projects to use this action, in all sorts of ways we can't predict ahead of time. There will be bugs and bad interactions with other tools, and we'll have to debug them — which is incredibly distracting, difficult, and time-consuming. Putting in additional work up front to systematically avoid 99.9% of those issues is necessary to ensure the success of the project: open-source projects die when they become too expensive to *maintain*.

Reply:
You're definitely right, `target` is quite a generic name, I'll change it to `semver-checks/target`.

I can easily prepare the test checking if the `target` directory exists and contains non-empty cache file (`registry-ref_slice-1_2_1.json` for the test repo). Is this sufficient? Or should I try to check for its validity in some way?